src/switcher.S

```diff
@@ -0,0 +1,134 @@
+#include "comps_offsets.h"
+
+.text
+.balign 4
+
+.global switcher_entry
+.global switch_compartment
+.global switch_compartment_end
+
+/**
+ * Entry point from user code to switcher function
+ *
+ * @param c0 DDC of switcher, containing compartment information
+ */
+.type switcher_entry, "function"
+switcher_entry:
+    mov      c29, c0
```

Comment:
Note that `c29` is callee-saved, so this should preserve it (since this is called from C code).

A simple fix for now would be to pass it in some caller-saved register not used for arguments (e.g. c10). However, I think we picked `c29` earlier because it's also special for `ldrblr` etc. Currently, we aren't using that, but maybe there's a development plan for it.

Reply:
That's a good point, having to save `c29`. Let's leave this conversation for the PR where the switcher is actually used. I'm sure there's more correctness issues to be discussed then.

Reply:
I've no objection in principle, but I worry that it'll be forgotten and bite us later. Bugs like these tend to go unnoticed because simple calling code might continue to work anyway.

The safer approach would be `s/c29/c10/` now (here and in `switch_compartment`). If we want to use `ldpblr` later, we can revisit this.

Reply:
I can do this for completion here, but the problem is I'd be needing to rebase those changes into my local branches, which are complete (and include `ldpblr` use appropriately). If you think it's worth it for having this PR more "complete", then I can do that, but this code is due to be overwritten shortly anyway. And I know I cut some corners regarding saving registers in the implementation I have locally, so that's something I definitely want to fix over in that PR; I definitely won't let it go.

Reply:
I'm not insisting on anything, so handle this as you prefer :-)